app/src/main/res/layout/fragment_register.xml

```diff
@@ -1,164 +1,200 @@
 <?xml version="1.0" encoding="utf-8"?>
-<layout xmlns:android="http://schemas.android.com/apk/res/android">
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
 
     <data>
 
         <variable
             name="viewModel"
             type="app.nexd.android.ui.auth.register.RegisterViewModel" />
+
+        <import type="android.view.View" />
     </data>
 
-    <ScrollView
+    <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:background="@android:color/white">
 
-        <RelativeLayout
+        <androidx.appcompat.widget.Toolbar
+            android:id="@+id/register_toolbar"
             android:layout_width="match_parent"
             android:layout_height="wrap_content"
-            android:orientation="vertical">
+            android:clickable="true"
+            android:focusable="true"
+            android:theme="@style/ToolbarThemeWhite"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:navigationIcon="?attr/homeAsUpIndicator"
+            android:visibility="@{ viewModel.alreadyRegistered ? View.INVISIBLE : View.VISIBLE }" />
```

Comment:
I think we should do this more properly by binding the toolbar to the navigation directly. It will then manage the back button automatically for us. Probably inlcuding it into MainActivity layout would even be the most consistent and easy solution.

Reply:
sure, we can use AppBarConfiguration for that. Do you want to show the toolbar on every screen now? Half of the screens in the design either have no toolbar or need a different theme for it.

Reply:
Unfortunately the designs are still not updated properly :( And since they are not consistent, we can use the easiest solution

Reply:
Do whatever you think reduces your efforts here. Because i don't know how the final toolbar will look like